src/Components/Components/src/Forms/ValidationSummary.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.RenderTree;
+using System;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /*
+     * Note: there's no reason why developers strictly need to use this. It's equally valid to
+     * put a @foreach(var message in context.GetValidationMessages()) { ... } inside a form.
+     * This component is for convenience only, plus it implements a few small perf optimizations.
+     */
+
+    /// <summary>
+    /// Displays a list of validation messages from a cascaded <see cref="EditContext"/>.
+    /// </summary>
+    public class ValidationSummary : ComponentBase, IDisposable
```

Comment:
I have some cool thoughts about this and other control-likes, but I'm refraining because I think this is all definitely good enough for us to give to users. 

One burning thought that I have is whether things like this should be sealed. I'm not sure that we know the implications of trying to write component classes that both do rendering and support inheritance well.

Reply:
Currently I'm thinking of subclassing as a very convenient way to customize rendering. Maybe this will turn out to be a mistake. I'd like to see if the usages justify it first. Let me know if you think you really do anticipate it to be a mistake!

Reply:
Ok, let's try this and see what happens. 

My expectation is that we'll get feedback from users who say **both**:

1. Please add feature XYZ to the base component because I want it
2. I don't want to subclass because it forces me to copy-paste your code and I don't want that either

However, I don't have a good answer for how we'd actually make this person happy. What you've done here makes it technically possible for this user to solve the problem themselves so that might be the best we can do.

Reply:
My hope is that, in the long term, it will be sufficient to have basic rendering of the standard HTML element corresponding to the input type (which we already have) with passthrough of arbitrary attributes (which we don't yet have). In the cases where someone wants to customize rendering beyond "pass through some more attributes", it's on them to either:

 * Subclass
 * Or, wrap the built-in input in some other component of their own